docs/setup-dev-environment.md

```diff
@@ -93,34 +101,91 @@ Restart postgres to pick up config changes.
 Create a superuser role for this user (`<username>`) if it does not already exist. The
 password `<pgpassword>` is arbitrary and probably should not be your Ubuntu login password.
 
-`sudo -u postgres psql -c "SELECT 1 FROM pg_user WHERE usename = '<username>';" | grep -q 1 || sudo -u postgres psql -c "CREATE ROLE <username> SUPERUSER LOGIN PASSWORD '<pgpassword>';"`
+```
+sudo -u postgres psql -c "SELECT 1 FROM pg_user WHERE usename = '<username>';" | grep -q 1 || sudo -u postgres psql -c "CREATE ROLE <username> SUPERUSER LOGIN PASSWORD '<pgpassword>';"
+```
+
+Set the `<pgpassword>` as an environment variable.
+
+```
+export PGPASSWORD=<pgpassword>
+```
 
 Create a colouring london database if none exists. The name (`<colouringlondondb>`) is arbitrary.
 
 ```
 sudo -u postgres psql -c "SELECT 1 FROM pg_database WHERE datname = '<colouringlondondb>';" | grep -q 1 || sudo -u postgres createdb -E UTF8 -T template0 --locale=en_US.utf8 -O <username> <colouringlondondb>
 ```
 
-To test the app user's connection to the database, you could run `psql` interactively:
+Run `psql` interactively.
 
 ```
 psql -d <colouringlondondb> -U <username> -h localhost
 ```
 
-Create the necessary postgres extensions.
+In `psql`, necessary postgres extensions.
 
 ```
-psql -d <colouringlondondb> -c "create extension postgis;"
-psql -d <colouringlondondb> -c "create extension pgcrypto;"
-psql -d <colouringlondondb> -c "create extension pg_trgm;"
+create extension postgis;
+create extension pgcrypto;
+create extension pg_trgm;
 ```
 
+Then quit `psql` by typing `\q` and hitting return.
+
 Now run all 'up' migrations to create tables, data types, indexes etc. The `.sql` scripts to
 do this are located in the `migrations` folder of your local repository.
 
-`ls ./colouring-london/migrations/*.up.sql 2>/dev/null | while read -r migration; do psql -d <colouringlondondb> < $migration; done;`
+`ls ./colouring-london/migrations/*.up.sql 2>/dev/null | while read -r migration; do psql -d <colouringlondondb> -U <username> -h localhost < $migration; done;`
+
+## Setting up Node
+
+Now upgrade the npm package manager to the most recent release with global privileges. This
+needs to be performed as root user, so it is necessary to export the node variables to the
+root user profile. Don't forget to exit from root at the end.
+
+```
+sudo su root
+export NODEJS_HOME=/usr/local/lib/node/node-v16.13.2/bin/
+export PATH=$NODEJS_HOME:$PATH
+npm install -g npm@latest
+exit
+```
+
+Now install the required Node packages. This needs to done from the `app` directory of your
+local repository, so that it can read from the `package.json` file.
+
+`cd ./colouring-london/app && npm install`
+
+## Loading the data
+
+<details>
+<summary> With a database dump </summary>
```

Comment:
@tomalrussell @matkoniecz I was wondering if you both did something similar to this when restoring the database from a dump file in a dev environment (virtualbox Ubuntu VM)? Essentially I ran through all the prior postgres steps in this doc ([see rendered md](https://github.com/colouring-cities/colouring-london/blob/docs-vagrant/docs/setup-dev-environment.md#loading-the-data)) up to this point, with the `<username>` and `<pgpassword>` set the same as the username and pass in the staging db, where I got the dump file (I'm unsure whether choosing the same username and password was neccesarry).

However the real problem (I think) seems to be that the VM quickly runs out of space, I set mine to have 10GB of storage for a 4GB dumpfile <- I thought this would be more than enough as one might expect the db to require less space than a file.

Not entirely sure that is the problem, I'm getting many repeats of this output: `invalid command \N` on running `psql -d <colouringlondondb> -U <username> -h localhost < <dumpfile>`, after it does some stuff, but it's hard to see which command failed as I can't scroll all the way up. Running it again results in:
```
^CERROR:  could not extend file "base/16385/17567": No space left on device
HINT:  Check free disk space.
```
hence I assume the space issue! Any help would be welcome. :)

Reply:
Ok, I tried everything again with a 50GB storage VM (dynamically allocated).

I tried again to create the db from the dump file I downloaded from staging server and mounted:

```
psql -d colouringlondondb -U cldbadmin -h localhost < db-dumps/colouringlondondb_staging_2022-02-14.dump
```

which results in the following error with `invalid command \N` repeated many times:

```
invalid command \N
invalid command \N
invalid command \.
ERROR:  syntax error at or near "3411839"
LINE 1: 3411839 osgb5000005210563591
        ^
invalid command \.
ERROR:  syntax error at or near "1"
LINE 1: 1 2019-10-03 05:33:00+00 /var/www/colouring-london/downloads...
        ^
invalid command \.
Killed
```

Perhaps I will try again with a different dump file.

Reply:
Or should I have created the db from the dumpfile before running the migrations files, perhaps that is the problem, I will check

Reply:
aha creating a fresh db with the dumpfile seems to have worked, although looks like I need to also add an extra user:

```
ERROR:  role "clwebapp" does not exist
```

Reply:
Yes, should be easiest to restore the dump to an empty database, but with the expected postgres users/roles created.

Reply:
Thanks @tomalrussell looks like I have that working, now facing the same issue with `libpq` as in the above thread, I'll try your idea!